### PR TITLE
fixed bug in collect_upstream_attributes

### PR DESCRIPTION
**Pull request by gcfang**
@phobson ensured all input for recfunctions.append_fields are strings instead of
unicodes

### CODE REVIEW

File: propagator/analysis.py [hunk 1]
```diff
@@ -663,6 +663,7 @@ def collect_upstream_attributes(subcatchments_table, target_subcatchments,
 
     final_cols = list(preserved_fields)
     final_cols.append(id_col)
+    utils._status('===ck1===', verbose=True, asMessage=True)
```
Comment (phobson) [inline propagator/analysis.py]: can we remove this?

**Comment by phobson** [comment]
Test coverage:

l_535e tmp

(single failure on extension context manager is irrelevant)

File: propagator/analysis.py (hunk 1 above)
Comment (gcfang) [inline propagator/analysis.py]: @phobson Removed in most recent commit

**Comment by phobson** [comment]
closes #22

**Pull request merged**